src/uct/ib/rc/verbs/rc_verbs_ep.c

```diff
@@ -135,6 +206,13 @@ ucs_status_t uct_rc_verbs_ep_put_short(uct_ep_h tl_ep, const void *buffer,
     UCT_CHECK_LENGTH(length, 0, iface->config.max_inline, "put_short");
 
     UCT_RC_CHECK_RES(&iface->super, &ep->super);
+    if ((buffer != NULL) || length || rkey) {
+        UCT_RC_VERBS_CHECK_FENCE(iface, ep);
+    } else {
+        /* "flush" operation - check for EP/IFACE resources only */
+        UCT_RC_CHECK_RES(&iface->super, &ep->super);
```

Comment:
should not we respect fence flag with flush as well?

Reply:
don't think so... flush doesn't has ordering - it just notify about all outstanding operations are completed

Reply:
flush should not outrun other ops - it breaks its semantic.

Reply:
ok, refactored